### DISCUSSION TITLE
Installing copilot globally on Mac

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [Copilot-Language/copilot#537] [Q&A]
I'm investigating how to install Copilot globally on Mac.

I'm trying to make the process trivial, automatic, and based on standard tools to install packages, and oriented towards non-haskellers. Having them manually interact with cabal is a no-go.

My goal is to eventually enable something along the following lines:
```
brew install copilot-language/copilot
```
after which all of the copilot modules should be available in GHC.

I can install ghc 9.8 (the latest version supported) with brew. I can also install cabal-install with brew (although it brings GHC 9.10, which is not ideal). I can then do `cabal update`, and `cabal install --lib copilot`. However, any attempts at adding that to a homebrew formula end up with the modules not being accessible or found.

One possibility would be to create a GHC environment (package environment? both?) in `/usr/local/Cellar/copilot/4.0_1/<whatever>` (as of the current version), and create wrappers around GHC that sets the right variables or passes the right arguments to GHC/GHCi (for example, `copilotc` and `copiloti`. I have not been successful at doing that yet. You can see my attempts at: https://github.com/ivanperez-keera/hombrew and specifically https://github.com/ivanperez-keera/hombrew/actions/workflows/mac-install.yaml.

Both GHC and cabal have changed a lot in how they manage packages in the last few versions, so this is non-trivial.

Any ideas?

**Comment by ivanperez-keera** [comment]
Tagging @RyanGlScott for awareness.

**Reply by RyanGlScott** [reply]
I've never used Homebrew much, so I'm not sure if I will be of much help here. That being said, I wonder if Homebrew has a convention where it only keeps files that are installed to certain locations, discarding all other files. I bring this up since most Haskell-based Homebrew formulae that I have seen install executables using `std_cabal_v2_args` (e.g., [this formula](https://github.com/Homebrew/homebrew-core/blob/6357209bb41850dbbf5f53571677a3fecd27ade0/Formula/c/cryptol.rb#L30)), and `std_cabal_v2_args` includes an `--installdir` argument (see [here](https://docs.brew.sh/Formula-Cookbook#std_cabal_v2_args)). Perhaps the environment file that `cabal install --lib` produces needs to be put in a similar location?

**Reply by ivanperez-keera** [reply]
Right. So, I am trying to place those files in Copilot's "own" location under homebrew: https://github.com/ivanperez-keera/hombrew/blob/master/copilot.rb#L13-L16

I think the problem is making sure that GHC creates those files correctly, and doesn't create any files needed to execute Copilot outside of that path.

It'd be easier if Copilot was an executable.
